### PR TITLE
Cover

### PR DESCRIPTION
**Pull request by matthieu-brillaxis**
Update of cover.
Each user from cover.json can be use now, we just need to define how we will use them to load data for the homepage.
Should we create a fake repo on a github account with a json file ?

### CODE REVIEW

**Comment by DavidBruant** [comment]
> Should we create a fake repo on a github account with a json file ?

No, we'll be using this repo. Please make a separate PR with a cover.json file listing real garment repos.

Is this PR ready to be merged?

File: src/components/pages/Home.vue [hunk 1]
```diff
@@ -36,21 +36,32 @@
...
-            console.log(response.status);
-            return;
+      const gh = new GitHub({
+        token: LoginStore.state.token,
+      });
+
+      const remoteRepo = gh.getRepo('ecvdbdx1617', 'github-for-fashion');
+      // const vm = this;         this will be usefull later
+      remoteRepo.getContents('master', 'content/cover.json', false).then((file) => {
+        let myJson = decodeURIComponent(escape(window.atob(file.data.content)));
```
Comment (DavidBruant) [inline src/components/pages/Home.vue]: ne need for `window.`

Comment (DavidBruant) [reply src/components/pages/Home.vue]: Unclear whether you need `decodeURIComponent` and `escape`

File: src/components/pages/Home.vue [hunk 2]
```diff
@@ -36,21 +36,32 @@
...
+        myJson = JSON.parse(myJson);
+        Object.keys(myJson).forEach((key) => {
```
Comment (DavidBruant) [inline src/components/pages/Home.vue]: By design, you already know the object has 2 keys and their names. No need to iterate over "all" keys

File: src/components/pages/Home.vue [hunk 3]
```diff
@@ -36,21 +36,32 @@
...
+          let user;
+          let repo;
+          if (key === 'primary') {
+            [user, repo] = String(myJson.primary).split('/');
+            console.log(user);
+            console.log(repo);
+            // We need to define the action we will execute with the user and repo from primary
+          } else {
+            Object.keys(myJson.secondary).forEach((secondaryKey) => {
+              [user, repo] = String(myJson.secondary[secondaryKey]).split('/');
```
Comment (DavidBruant) [inline src/components/pages/Home.vue]: Why the `String` conversion?

Comment (matthieu-brillaxis) [reply src/components/pages/Home.vue]: If I don't do string conversion the .split() is not working like I want.

**Comment by matthieu-brillaxis** [comment]
I prefer to wait before merging this PR. We can watch that at the next class. I will try to make changes before the next class since I know your comments.

File: src/components/pages/Home.vue (hunk 1 above)
Comment (matthieu-brillaxis) [reply src/components/pages/Home.vue]: Ok

File: src/components/pages/Home.vue [hunk 4]
```diff
@@ -36,21 +35,32 @@
...
+        primaryRepo.getDetails().then((response) => {
+          const primaryGarment = response.data;
+          this.mainCard.title = primaryGarment.name;
+          this.mainCard.artist = primaryGarment.owner.login;
+          this.mainCard.description = primaryGarment.description;
+        });
+        primaryRepo.getContributorStats().then((response) => {
+          const primaryGarment = response.data[0];
+          this.mainCard.contributor = primaryGarment.total;
+        });
```
Comment (DavidBruant) [inline src/components/pages/Home.vue:51]: secundary missing

File: src/components/pages/Home.vue [hunk 5]
```diff
@@ -36,21 +35,32 @@
...
+
+      const remoteRepo = gh.getRepo('ecvdbdx1617', 'github-for-fashion');
+      remoteRepo.getContents('master', 'content/cover.json', false).then((file) => {
+        let myJson = atob(file.data.content);
+        myJson = JSON.parse(myJson);
+        /* eslint-disable prefer-const */
+        /* eslint-disable no-unused-vars */
+        let user = '';
+        let repo = '';
+        [user, repo] = String(myJson.primary).split('/');
```
Comment (DavidBruant) [inline src/components/pages/Home.vue]: string conversion?

File: src/components/pages/Home.vue [hunk 6]
```diff
@@ -36,21 +35,32 @@
...
+      const gh = new Github({
+        token: LoginStore.state.token,
+      });
+
+      const remoteRepo = gh.getRepo('ecvdbdx1617', 'github-for-fashion');
+      remoteRepo.getContents('master', 'content/cover.json', false).then((file) => {
+        let myJson = atob(file.data.content);
+        myJson = JSON.parse(myJson);
+        /* eslint-disable prefer-const */
+        /* eslint-disable no-unused-vars */
```
Comment (DavidBruant) [inline src/components/pages/Home.vue]: re-enable these rules

**Review by DavidBruant — CHANGES_REQUESTED** [review]
LGTM otherwise

**Pull request merged**